### PR TITLE
docs: add porting guide

### PR DESCRIPTION
**Pull request by katport**
## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: docs/tutorials/porting-tutorial.txt [hunk 1]
```diff
@@ -0,0 +1,473 @@
...
+        momentum: 0.9
+        weight_decay: 1e-4
+        pretrained: True
+    records_per_epoch: 60000
+    searcher:
+        name: single
+        metric: val_loss
+        smaller_is_better: false
+        max_length:
+            Epochs: 10
```
Comment (mluyo3414) [inline docs/tutorials/porting-tutorial.txt:89]: should be epochs in lower case

File: docs/tutorials/porting-tutorial.txt [hunk 2]
```diff
@@ -0,0 +1,473 @@
...
+====
+Before we begin, we need to create our core files. Determined requires two files to be defined: the model definition and the experiment configuration.
+
+Model Definition
+++++++++++++++++
+The model definition contains the trial class, which contains the model definition and training loop. Your deep learning framework defines which ``Determined.Trial()`` class to inherit. In our case, we will be working with PyTorch, which means we will be working with :class:`determined.pytorch.PyTorchTrial`. Once we inherit this class, we will be required to override specific functions that represent a training script’s core components. When starting a new port, it is helpful to work off of a skeleton to keep track of what is still required. A good starting template can be found below: 
+
+.. code:: python
+
+    From typing import Any, Dict
```
Comment (mluyo3414) [inline docs/tutorials/porting-tutorial.txt:38]: missing some imports I believe 
`from typing import Any, Dict, Union, Sequence`

File: docs/tutorials/porting-tutorial.txt (hunk 1 above)
Comment (katport) [reply docs/tutorials/porting-tutorial.txt]: nice catch

**Review by naren-determined — APPROVED** [review]
Lgtm assuming it passes linting tests!

**Pull request merged**